lib/ddtrace/contrib/rails/framework.rb

```diff
@@ -80,21 +80,32 @@ def self.configure(config)
         end
 
         def self.auto_instrument_redis
-          # configure Redis PIN
-          return unless (defined? ::Rails.cache) && ::Rails.cache.respond_to?(:data)
-          pin = Datadog::Pin.get_from(::Rails.cache.data)
-          return unless pin
+          return unless ::Rails.configuration.datadog_trace[:auto_instrument_redis]
+          Datadog::Tracer.log.debug('Enabling auto-instrumentation for Redis client')
 
-          # enable Redis instrumentation if activated
-          pin.tracer = nil unless ::Rails.configuration.datadog_trace[:auto_instrument_redis]
-          return unless pin.tracer
-          Datadog::Tracer.log.debug("'redis' module found, Datadog 'redis' integration is available")
+          # patch the Redis library and reload the CacheStore if it was using Redis
+          Datadog::Monkey.patch_module(:redis)
+
+          # reload the cache store if it's available and it's using Redis
+          return unless defined?(::ActiveSupport::Cache::RedisStore) &&
+                        defined?(::Rails.cache) &&
+                        ::Rails.cache.is_a?(::ActiveSupport::Cache::RedisStore)
+          Datadog::Tracer.log.debug('Enabling auto-instrumentation for redis-rails connector')
+
+          # backward compatibility: Rails 3.x doesn't have `cache=` method
```

Comment:
Rails 3.x relies in a `RAILS_CACHE` global; Rails > 3 deprecated this global in favor of a plain ``Rails.cache``